libc/intrin/kprintf.h

```diff
@@ -9,25 +9,49 @@
 #define kvsnprintf   __kvsnprintf
 #define kloghandle   __kloghandle
 #define kisdangerous __kisdangerous
+#define uprintf      __uprintf
+#define uvprintf     __uvprintf
 
 #if !(__ASSEMBLER__ + __LINKER__ + 0)
-COSMOPOLITAN_C_START_
+#include "libc/dce.h"
 
-void kprintf(const char *, ...);
-size_t ksnprintf(char *, size_t, const char *, ...);
-void kvprintf(const char *, va_list);
-size_t kvsnprintf(char *, size_t, const char *, va_list);
-bool kisdangerous(const void *);
+COSMOPOLITAN_C_START_
 
 void kprintf(const char *, ...);
 size_t ksnprintf(char *, size_t, const char *, ...);
 void kvprintf(const char *, va_list);
 size_t kvsnprintf(char *, size_t, const char *, va_list);
 
 bool kisdangerous(const void *);
+
 void klog(const char *, size_t);
+void _klog_serial(const char *, size_t);
 long kloghandle(void);
 
+void uprintf(const char *, ...);
+void uvprintf(const char *, va_list);
+
+#define KINFOF(FMT, ...)                                      \
```

Comment:
could you use `#ifndef TINY` for these instead and remove the include

Reply:
It's mostly an issue for header files. I try to keep include complexity down. It also made your macros more elegant.